site/app/controllers/UserProfileController.php

```diff
@@ -166,4 +167,36 @@ public function changeProfilePhoto() {
             }
         }
     }
+
+    /**
+     * @Route("/user_profile/change_secondary_email", methods={"POST"})
+     * @return JsonResponse
+     * @throws \ImagickException
+     */
+    public function changeSecondaryEmail() {
+        $user = $this->core->getUser();
+
+        if (!empty($_POST['secondary_email']) && !empty($_POST['secondary_email_notify'])) {
+            $secondaryEmail = trim($_POST['secondary_email']);
+            $secondaryEmailNotify = trim($_POST['secondary_email_notify']);
+
+            if ($user->validateUserData('user_email_secondary', $secondaryEmail) === true) {
+                $user->setSecondaryEmail($secondaryEmail);
+                $user->setEmailBoth($secondaryEmailNotify === "true");
+                $user->setUserUpdated(true);
```

Comment:
Reading the docstring for the `user_updated` field

>This flag is set TRUE when a user edits their own preferred firstname.  When TRUE, preferred firstname
     *           is supposed to be locked from changes via student auto feed script.  Note that auto feed is still
     *           permitted to change (correct?) a user's legal firstname/lastname and email address.
     
    It sounds like it should only be set to true for when a user updates their preferred name, unless we also want a similar behavior for a secondary email which I don't think so since that information does not come from the autofeed source and can't clobber anything